caikit_nlp/modules/text_generation/text_generation_local.py

```diff
@@ -614,50 +625,10 @@ def _launch_training(
 
 def get(train_stream):
     for data in train_stream:
-        yield {"input": data.input, "output": data.output}
-
-
-def tokenize_function_seq2seq(
-    example: GenerationTrainRecord,
-    tokenizer: "AutoTokenizer",
-    max_source_length: int,
-    max_target_length: int,
-):
-    """Tokenization function to be used for seq2seq training; this function consumes a
-    GenerationTrainRecord object and applies the verbalizer to it followed by
-    the model tokenizer. Finally, we postprocess by ignoring pad tokens in the label IDs.
-
-    Args:
-        example: GenerationTrainRecord
-            Training data model object to convert a form we can learn on.
-
-    Returns:
-        transformers.tokenization_utils_base.BatchEncoding
-            encoded tokenization output corresponding to the input example.
-    """
-    IGNORE_ID = -100
-    # Render the verbalizer template with the attributes of this data model example
-    source = example["input"]
-
-    targets = example["output"]
-    model_inputs = tokenizer(
-        source,
-        max_length=max_source_length,
-        padding="max_length",
-        truncation=True,
-    )
-    labels = tokenizer(
-        targets,
-        max_length=max_target_length,
-        padding="max_length",
-        truncation=True,
-    )
-
-    labels = labels["input_ids"]
-
-    labels = list(
-        map(lambda x: IGNORE_ID if x == tokenizer.pad_token_id else x, labels)
-    )
-    model_inputs["labels"] = labels
-
-    return model_inputs
+        # Handle token unwrapping for causal language modeling
```

Comment:
nit: should we rename the function name and may be move it to somewhere else? am thinking since this is specially for training record,it might make sense to move it to datamodel somewhere 🤔

Reply:
Yeah, I think you're right! Once I have it working correctly with causal LM, I can put it in the dm or toolkit, that is probably better